### ISSUE TITLE
Expose the current issuer

### ISSUE THREAD

**Issue by mrkvon** [inrupt/solid-client-authn-js#2153]
### Search terms you've used
issuer

### Impacted environment

In which environment would the proposed feature apply ?

- [x] The browser
- [x] Node.js
- [ ] Other (please specify): ...
- [ ] I'm not sure.

### Feature suggestion

> [...] "retrieve the current issuer from the session" [and] expose that via the ISessionInfo interface, potentially

_per @ThisIsMissEm's suggestion in a PM_

In certain cases, it may be useful for an App to know the OIDC issuer which was used to authenticate current user.

E.g. a user with webId `https://someuser.solidcommunity.net/profile/card#me` may have an issuer `https://solidcommunity.net`, and we want to know that.

### Expected functionality/enhancement

Currently, [ISessionInfo](https://docs.inrupt.com/developer-tools/api/javascript/solid-client-authn-browser/interfaces/ISessionInfo.html) doesn't expose issuer, and it could.
The issuer is exposed within the Authorization header of the [fetch function](https://docs.inrupt.com/developer-tools/api/javascript/solid-client-authn-browser/functions.html#fetch) as `iss` property of a JWT token. It's also exposed in localStorage, however i'm told this is going to change.

### Actual functionality/enhancement

Expose the current issuer in ISessionInfo (or in other standard way)

### Use Cases

- As of now, some OIDC Identity Providers (typically https://solidcommunity.net and other NSS-based servers) don't include [`solid:oidcIssuer`](https://solid.github.io/solid-oidc/#oidc-issuer-discovery) in personal profile documents. This is sufficient for authenticating into NSS Pods, but fails when authenticating using e.g. [@solid/access-token-verifier](https://github.com/CommunitySolidServer/access-token-verifier).
So, sometimes an app needs to fix such incomplete profile document (if it can), to make it compliant. Hence, it needs to know the issuer.

- Also, perhaps an app wants to show the current issuer in UI... 🤷🏾‍♀️ 

---

ping @jeff-zucker

**Comment by jeff-zucker** [comment]
On a server compliant with the OIDC spec, it is impossible to login unless there is a solid:oidcIssuer predicate in the profile.  So it is impossible for an app to login with OIDC and  grab the issuer from the login unless there is already an issuer in the profile.  The only reason this works on NSS is that it is both IdP and Resource provider and auto-assigns itself as the issuer.  NSS has already been patched to include the oidcIssuer triple in all new profiles.  If you want to fix a missing NSS issuer on older profiles, the issuer will always be NSS.

**Comment by ThisIsMissEm** [comment]
@jeff-zucker true, you could look at the profile, but if the profile includes multiple `solid:oidcIssuer` predicates, then you'd need some way to figure out which was used to authenticate the current session. We do already keep track of this internally, it's just not exposed as a public API: https://github.com/inrupt/solid-client-authn-js/blob/main/packages/core/src/sessionInfo/ISessionInfo.ts#L71

**Comment by jeff-zucker** [comment]
The spec (which should be ready for public review real soon now) currently says about apps wanting to login - 
* 0 issuers  : inform user their profile is broken
* 1 issuer : redirect there for login
* >1 issuer : offer the user a choice of where to login

Some sort of "preferredIssuer" predicate would be useful in this situation.

**Comment by ThisIsMissEm** [comment]
Yeah, but once you are logged in, being able to find out which issuer logged you in could be useful; we already keep track of it, it'd just be a matter of exposing it as a public value.

**Comment by jeff-zucker** [comment]
Ah, I see, yes an app might want to know that.

**Comment by mrkvon** [comment]
> NO, don't do that.

So, do you suggest to only fix [well-known NSS](https://solidproject.org/users/get-a-pod#get-a-pod-from-a-pod-provider) profiles? Or perhaps to give up on all older NSS profiles? The former sounds like a sensible compromise. The latter would be rather unsatisfying...

> An issuer does not have the trailing slash

I've seen trailing slashes on solidweb.me (CSS). E.g. see https://solidweb.me/test/profile/card

**Comment by jeff-zucker** [comment]
> I've seen trailing slashes on solidweb.me (CSS). E.g. see https://solidweb.me/test/profile/card

The [OIDC spec](https://solidproject.org/TR/oidc) shows the oidcIssuer with no slash but does not specifically say that it should be omitted.   @acoburn - is solidweb.me wrong to include the trailing slash  on the oidcIssuer triple?
